File: src/Pokedex.Api/Dtos/PokemonDto.cs
namespace Pokedex.Api.Dtos;

public record class PokemonDto
{
    public int Number { get; set; }
    public string Name { get; set; }
    public string PrimaryType { get; set; }
    public string? SecondaryType { get; set; }
    public float CatchRate { get; set; }
    public bool IsLegendary { get; set; }
    public string? Avatar { get; set; }
    public PhysicalAttributesDto PhysicalAttributes { get; set; }
    public GenderRatioDto GenderRatio { get; set; }  
    public BaseStatsDto? BaseStats { get; set; }
    public PokemonDto? EvolvesTo { get; set; }
    public IEnumerable<PokemonDto>? EvolvesFrom { get; set; }

}

File: src/pokedex.spa/src/models/PokemonDto.ts
export default interface PokemonDto {
	name: string;
	avatar: string;
	baseStats: BaseStatsDto;
	catchRate: number;
	evolvesFrom: PokemonDto[];
	evolvesTo: PokemonDto;
	genderRatio: GenderRatioDto;
	isLegendary: boolean;
	number: number;
	physicalAttributes:PhysicalAttributesDto;
	primaryType: string;
	secondaryType: string;
}

export interface BaseStatsDto {
	hP: number;
	attack: number;
	defense: number;
	specialAttack: number;
}

export interface GenderRatioDto {
	femaleRatio: number;
	maleRatio: number;
}

export interface PhysicalAttributesDto {
	height: number;
	weigth: number;
}